### PR TITLE
New Linting Setup

### PR DESCRIPTION
**Pull request by JeffThorslund**
Ok I got the linting set up. Here is how it works. 

Setup is heavily based on this one. 

https://github.com/wesbos/eslint-config-wesbos

- `.eslintrc` contains all the setup and import. 
- `npm run lint` shows a list of all linting errors.
- `npm run lint:fix` fixes auto-fixable errors and shows errors it cant fix. 

**IF YOU WANT TO SET IT UP SO IT USES OUR SETTINGS WHEN YOU SAVE**

Follow the `With VS Code` section of the github link I posted at the top. It shows how to set up the default autosaver. Even if that doesnt work, you still have access to `npm run lint` and `npm run lint:fix`, which you should run before a commit or push. 

Linting is now a lot stricter but we can take out annoying rules when we come across them. 

Dont worry about reading all the code changes except the first one, `d4662c1`

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
+  "extends": ["airbnb", "prettier", "prettier/react"],
```
Comment (bekatru) [inline .eslintrc:2]: airbnb? :)

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      "warn",
+      {
+        "aspects": ["invalidHref"]
+      }
+    ],
+    "react-hooks/rules-of-hooks": "error",
+    "react-hooks/exhaustive-deps": "warn"
+  },
+  "plugins": ["html", "prettier", "react-hooks"]
+}
```
Comment (bekatru) [inline .eslintrc:135]: Wow, this is big!

File: client/src/components/_utils/index.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
-import useToggle from "./useToggle";
+import useToggle from './useToggle';
```
Comment (bekatru) [inline client/src/components/_utils/index.js:1]: Single quotes! I like it!

File: client/src/components/general/SearchBar.js [hunk 4]
```diff
@@ -1,38 +1,24 @@
...
+  >
+    <Form.Group controlId="search-bar">
+      <Form.Control
+        value={value}
+        onChange={handleChange}
+        placeholder={placeholder}
+      />
+    </Form.Group>
+  </Form>
+);
```
Comment (bekatru) [inline client/src/components/general/SearchBar.js:22]: Is this done by linting too?

**Review by bekatru — APPROVED** [review]
This makes a big difference and a really good job! Is is gonna be installed if I pull these changes?

**Comment by JeffThorslund** [comment]
Sweet, I'll handle the merge between your branch and this linting.

File: .eslintrc (hunk 1 above)
Comment (JeffThorslund) [reply .eslintrc:2]: Yaa Airbnb developers have created the golden standard of linting. Its a very common linting setup to use.

**Pull request merged**